src/CommunityToolkit.Maui/Views/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,176 @@
+	protected override void OnPropertyChanged(string propertyName)
+	{
+		if (propertyName is not null && (propertyName == CacheValidityProperty.PropertyName || propertyName == CachingEnabledProperty.PropertyName))
+		{
+			OnUriChanged();
+		}
+
+		base.OnPropertyChanged(propertyName);
+	}
+
+	/// <summary>Gets or sets a <see cref="TimeSpan"/> structure that indicates the length of time after which the image cache becomes invalid.</summary>
+	public TimeSpan CacheValidity
+	{
+		get => (TimeSpan)GetValue(CacheValidityProperty);
+		set => SetValue(CacheValidityProperty, value);
+	}
+
+	/// <summary>Gets or sets a Boolean value that indicates whether caching is enabled on this <see cref="GravatarImageSource"/> object.</summary>
+	public bool CachingEnabled
+	{
+		get => (bool)GetValue(CachingEnabledProperty);
+		set => SetValue(CachingEnabledProperty, value);
+	}
+
+	/// <summary>Gets or sets the email address.</summary>
+	public string? Email
+	{
+		get => (string)GetValue(EmailProperty);
+		set => SetValue(EmailProperty, value);
+	}
+
+	/// <summary>Gets or sets the default image.</summary>
+	public DefaultImage Image
+	{
+		get => (DefaultImage)GetValue(ImageProperty);
+		set => SetValue(ImageProperty, value);
+	}
+
+	/// <summary>Gets a value indicating whether the control URI is empty.</summary>
+	public override bool IsEmpty => Uri == null;
+
+	/// <summary>Gets or sets the URI for the image to get.</summary>
+	[System.ComponentModel.TypeConverter(typeof(UriTypeConverter))]
+	public Uri Uri { get; private set; }
+
+	/// <summary>Gets or sets the control size property.</summary>
+	int GravatarSize
+	{
+		get => gravatarSize;
+		set => gravatarSize = Math.Clamp(value, 1, 2048);
+	}
+
+	/// <summary>Returns the Uri as a string.</summary>
+	/// <returns>String of the URI.</returns>
+	public override string ToString()
+	{
+		return $"Uri: {Uri}, Email:{Email}, Size:{GravatarSize}, Image: {DefaultGravatarName(Image)}, CacheValidity:{CacheValidity}, CachingEnabled:{CachingEnabled}";
+	}
+
+	/// <summary>On parent set.</summary>
+	protected override void OnParentSet()
+	{
+		object element = Parent;
+		base.OnParentSet();
+		if (element is Microsoft.Maui.Controls.Image parentImage)
+		{
+			var height = parentImage.Height >= 0 ? parentImage.Height : Math.Max(parentImage.HeightRequest, GravatarImageSourceDefaults.DefaultSize);
+			var width = parentImage.Width >= 0 ? parentImage.Width : Math.Max(parentImage.WidthRequest, GravatarImageSourceDefaults.DefaultSize);
+			var sizeFromParent = (int)Math.Max(width, height);
+			if (GravatarSize != sizeFromParent)
+			{
+				GravatarSize = sizeFromParent;
+				HandleNewUriRequested(Email, Image);
+			}
+		}
+	}
+
+	static string DefaultGravatarName(DefaultImage defaultGravatar)
+			=> defaultGravatar switch
+			{
+				DefaultImage.FileNotFound => "404",
+				DefaultImage.MysteryPerson => "mp",
+				_ => $"{defaultGravatar}".ToLower(),
+			};
+
+	static string GetMd5Hash(string? str)
```

Comment:
Can we change this method to use `ReadOnlySpan<char>` instead of `string`? [Here is the version](https://gist.github.com/pictos/2df3ced9b4d78c58c5c5133940715a5b#file-gravatarimagesourcesilly-cs-L51-L61) with the `ReadOnlySpan`

Reply:
Implemented.